Makefile

```diff
@@ -22,7 +22,7 @@ run: vet
 .PHONY: mocks
 mocks: clean-mocks
 	@echo "=============generating mocks============="
-	grep -rl --exclude "./vendor/*" --include "*.go" "interface {" . | while read -r file ; do mockgen --source=$$file --destination mocks/$$file ; done
+	grep -rl --exclude-dir 'vendor' --exclude-dir 'data' --include "*.go" "interface {" . | while read -r file ; do mockgen --source=$$file --destination mocks/$$file ; done
```

Comment:
Possibly, I haven't checked.

I made this change while I was trying to fix the codecov reporting. It didn't make any difference for the reported code coverage but it did clean up the logs so I thought I'd just leave it in.

But yeah it could solve the issue with GH Actions as well because the CI script was failing due to an error when generating mocks and with this change we shouldn't get that error anymore. Though I'm not completely certain it would fix the problem either because GH Actions CI was working previously while the mocks script was still throwing errors. I'll look into this when I'm working on #79.